src/runtimes/node/finder.ts

```diff
@@ -62,11 +65,13 @@ const getFunctionAtPath = async function (srcPath: string): Promise<SourceFile |
     return
   }
 
+  const iscValues = await findISCDeclarationsInPath(mainFile)
```

Comment:
One thing to consider is that `listFunctions` is [called by Netlify Build](https://github.com/netlify/build/blob/088ff1bba18be6e49447ba2b250f1cb5f5d3af22/packages/build/src/plugins_core/functions/utils.js#L23) as part of the build process, in order to show a list of functions that will get bundled.

In those circumstances, getting access to the in-source config declarations is not needed, and the additional effort of parsing the main files will be wasted.

How do you feel about adding an additional parameter to `listFunctions`, disabled by default, which determines whether to extract ISC?

Also, I'm not sure that logic should be part of `getFunctionAtPath`, because finding whether a function exists and cracking it open to extract in-source config feel like two separate concerns. I'd prefer a similar approach to how we're [doing it now](https://github.com/netlify/zip-it-and-ship-it/blob/main/src/runtimes/node/index.ts#L120), where we augment the function object with ISC data (which in the case of `listFunctions` should happen conditionally).

Reply:
implemented in 4048c42. is that what you had in mind?

Reply:
You mean https://github.com/netlify/zip-it-and-ship-it/pull/899/commits/15c1cadfe33854a8bee6810cd66923f22f22a089? If so, yes! I left some minor comments below.